lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -91,7 +63,7 @@ def disconnect
     #
     # @return [Packet|nil] A Packet of consisting of the bytes read from the
     #   stream.
-    def read
+    def read_data
```

Comment:
read_data is now the lowest level method which should go out and get the raw data